File: cpp/abc088_b.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

int main()
{
  int N; cin >> N;
  vector<int> A(N);
  for(int i=0; i<N; i++) cin >> A[i];

  // 数列を昇順にソート
  sort(A.begin(), A.end());
  // 数列を降順にソート
  reverse(A.begin(), A.end());

  // （先手が取るものの総和） - （後手が取るものの総和）を計算
  int ans = 0;
  for(int i=0; i<N; i++) {
    if (i % 2 == 0) {
      // 偶数番目なら先手が取る
      ans += A[i];
    } else {
      // 奇数番目なら後手が取る
      ans -= A[i];
    }
  }
  cout << ans << endl;
  return 0;
}

File: go/abc088_b.go
package main

import (
  "bufio"
  "fmt"
  "os"
  "sort"
  "strconv"
  "strings"
)

func main() {
  var n int
  fmt.Scan(&n)

  stdin := bufio.NewScanner(os.Stdin)
  stdin.Scan()
  s := stdin.Text()

  split := strings.Split(s, " ")
  intList := convStringListToIntList(split)

  sort.Sort(sort.Reverse(sort.IntSlice(intList)))
  alice, bob := assignToAliceAndBob(intList)

  aliceSum := sum(alice)
  bobSum := sum(bob)
  fmt.Println(aliceSum - bobSum)
}

func convStringListToIntList(stringList []string) []int {
  intList := make([]int, len(stringList))
  for i, v := range stringList {
    iv, _ := strconv.Atoi(string(v))
    intList[i] = iv
  }
  return intList
}

func assignToAliceAndBob(list []int) ([]int, []int) {
  aliceList := make([]int, len(list))
  bobList := make([]int, len(list))
  c := 0
  for i, v := range list {
    if i%2 == 0 {
      aliceList[c] = v
    } else {
      bobList[c] = v
    }
    c += 1
  }
  return aliceList, bobList
}

func sum(list []int) int {
  var result int
  for _, v := range list {
    result += v
  }
  return result
}
